### ISSUE TITLE
[MIRA] Sometimes converting index to mira type (after in-memory replay) takes forever (Write Stalls).

### ISSUE THREAD

**Issue by Gandalf-the-Grey** [steemit/steem#3501]
Literally. It will **never** end.
Ubuntu 18.04 LTS
steemd: `v0.22.1`
RocksDB: `6.0.2`

Dedicated machine, enough RAM to avoid using swap, storage on 2x NVMe (RAID1)

In my case it was `rocksdb_vesting_delegation_object`.

Symptoms:
It's running for over 42 hours now (usually whole replay would finish with 4x less resources)
`ls -1 ~/.steemd/blockchain/rocksdb_vesting_delegation_object | wc -l`
```
204358
```
`fgrep WARN ~/.steemd/blockchain/rocksdb_vesting_delegation_object/LOG | tail -n 6`
```
2019/09/05-08:13:21.645897 7fd0d17fa700 [WARN] [db/column_family.cc:738] [boost::mpl::v_item<steem::chain::by_id, boost::mpl::vector0<mpl_::na>, 0>] Stopping writes because we have 2 immutable memtables (waiting for flush), max_write_buffer_number is set to 2
2019/09/05-08:13:22.332611 7fd1cafdcf00 [WARN] [db/column_family.cc:738] [boost::mpl::v_item<steem::chain::by_delegation, boost::mpl::vector0<mpl_::na>, 0>] Stopping writes because we have 2 immutable memtables (waiting for flush), max_write_buffer_number is set to 2
2019/09/05-08:13:22.340063 7fd068ff1700 [WARN] [db/column_family.cc:738] [boost::mpl::v_item<steem::chain::by_delegation, boost::mpl::vector0<mpl_::na>, 0>] Stopping writes because we have 2 immutable memtables (waiting for flush), max_write_buffer_number is set to 2
2019/09/05-08:13:23.537544 7fd1cafdcf00 [WARN] [db/column_family.cc:738] [boost::mpl::v_item<steem::chain::by_id, boost::mpl::vector0<mpl_::na>, 0>] Stopping writes because we have 2 immutable memtables (waiting for flush), max_write_buffer_number is set to 2
2019/09/05-08:13:23.550014 7fd0d17fa700 [WARN] [db/column_family.cc:738] [boost::mpl::v_item<steem::chain::by_id, boost::mpl::vector0<mpl_::na>, 0>] Stopping writes because we have 2 immutable memtables (waiting for flush), max_write_buffer_number is set to 2
2019/09/05-08:13:23.825262 7fd173fff700 [WARN] [db/column_family.cc:738] [boost::mpl::v_item<steem::chain::by_id, boost::mpl::vector0<mpl_::na>, 0>] Stopping writes because we have 2 immutable memtables (waiting for flush), max_write_buffer_number is set to 2
```

TL;DR:
`Stopping writes because we have 2 immutable memtables (waiting for flush), max_write_buffer_number is set to 2`

> **Too many memtables**.
> When the number of memtables waiting to flush is greater or equal to `max_write_buffer_number`, writes are fully stopped to wait for flush finishes. In addition, if `max_write_buffer_number` is greater than `3`, and the number of memtables waiting for flush is greater than or equal to `max_write_buffer_number - 1`, writes are stalled.

Worth trying:
Increase `max_background_flushes` to have more flush threads.
Increase `max_write_buffer_number` to have smaller memtable to flush.

References:
[RocksDB Write Stalls](https://github.com/facebook/rocksdb/wiki/Write-Stalls)

**Comment by mvandeberg** [comment]
While implementing platform independent state files, we made optimizations to this process and were able to improve conversion times by around 50% in testing. I am going to close this issue, if times need optimized further, please open a new issue.
